homeassistant/components/yolink/device_trigger.py

```diff
@@ -0,0 +1,86 @@
+"""Provides device triggers for YoLink."""
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+from yolink.const import ATTR_DEVICE_SMART_REMOTER
+
+from homeassistant.components.device_automation import DEVICE_TRIGGER_BASE_SCHEMA
+from homeassistant.components.homeassistant.triggers import event as event_trigger
+from homeassistant.const import CONF_DEVICE_ID, CONF_DOMAIN, CONF_PLATFORM, CONF_TYPE
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.trigger import TriggerActionType, TriggerInfo
+from homeassistant.helpers.typing import ConfigType
+
+from . import DOMAIN
+from .const import YOLINK_EVENT
+
+CONF_BUTTON_1 = "button_1"
+CONF_BUTTON_2 = "button_2"
+CONF_BUTTON_3 = "button_3"
+CONF_BUTTON_4 = "button_4"
+CONF_SHORT_PRESS = "short_press"
+CONF_LONG_PRESS = "long_press"
+
+TRIGGER_SCHEMA = DEVICE_TRIGGER_BASE_SCHEMA.extend({vol.Required(CONF_TYPE): str})
```

Comment:
We normally define the allowed trigger types as a set and use `vol.In` instead of `str` as validator.